app/models/mixins/collections_parser_mixin.rb

```diff
@@ -23,12 +30,12 @@ def safe_call
     yield
   rescue Excon::Errors::Forbidden => err
     # It can happen user doesn't have rights to read some tenant, in that case log warning but continue refresh
-    _log.warn "Forbidden response code returned in provider: #{@os_handle.address}. Message=#{err.message}"
+    _log.warn "Forbidden response code returned in provider: #{(@os_handle || @ems).address}. Message=#{err.message}"
```

Comment:
this mixin depends on `@options`, `@data`, `@data_index`, `@os_handle` and `@ems`